snmp/datadog_checks/snmp/pysnmp_types.py

```diff
@@ -51,16 +42,40 @@
     'ObjectName',
     'ObjectType',
     'OctetString',
+    'Opaque',
     'SnmpEngine',
     'UdpTransportTarget',
     'usmDESPrivProtocol',
     'usmHMACMD5AuthProtocol',
     'UsmUserData',
-    'ZeroBasedCounter64',
-    'Counter32',
-    'Counter64',
-    'Gauge32',
-    'Unsigned32',
-    'Integer',
-    'Integer32',
 ]
+
+# Additional types that are not part of the SNMP protocol (see RFC 2856).
+CounterBasedGauge64, ZeroBasedCounter64 = MibBuilder().importSymbols(
+    'HCNUM-TC', 'CounterBasedGauge64', 'ZeroBasedCounter64'
+)
+
+# SNMP value types that we explicitly support.
+PYSNMP_COUNTER_CLASSES = {
+    Counter32.__name__,
+    Counter64.__name__,
+    ZeroBasedCounter64.__name__,
+}
+PYSNMP_GAUGE_CLASSES = {
+    Gauge32.__name__,
+    CounterBasedGauge64.__name__,
+    Integer.__name__,
+    Integer32.__name__,
+    Unsigned32.__name__,
```

Comment:
I wonder if we could get rid of the class import. Those names are static and always match the class names, what do you think of using them directly? It's particularly dumb for the additional types that we import by names, to then only refer to them by names.

Reply:
So, I actually like that we import classes from PySNMP. I think it's important we keep that "this comes from PySNMP code" relationship. But the import/reference/delete usage definitely looks odd. I'll push a commit so that these sets contain actual classes, and then we access `.__name__` at compare time in `metrics.py` — probably sounds less clunky?

Reply:
Changed to hardcoded strings in latest commit. Turns out the comment that's been there about additional types is correct: `isinstance()`/`isSameTypeWith()` are unreliable for some types, so we really need to stick to string comparisons.